src/compile/data.ts

```diff
@@ -165,12 +166,12 @@ export namespace source {
         }
 
         transform.push(binTrans);
-        if (scaleType(fieldDef, channel, model.mark()) === 'ordinal') {
+        if (scaleType(fieldDef, channel, model.mark()) === 'ordinal' || channel === COLOR) {
```

Comment:
Probably worth add comment that `COLOR` ramp has `linear` type for the main scale.